/*
 * This file was automatically generated by EvoSuite
 * Fri Feb 21 06:51:22 GMT 2020
 */

package org.apache.groovy.util;

import org.junit.Test;
import static org.junit.Assert.*;
import static org.evosuite.shaded.org.mockito.Mockito.*;
import static org.evosuite.runtime.EvoAssertions.*;
import groovy.lang.GroovyRuntimeException;
import java.util.concurrent.ExecutorService;
import java.util.concurrent.Executors;
import java.util.concurrent.Future;
import java.util.concurrent.TimeUnit;
import org.apache.groovy.util.JavaShell;
import org.evosuite.runtime.EvoRunner;
import org.evosuite.runtime.EvoRunnerParameters;
import org.evosuite.runtime.ViolatedAssumptionAnswer;
import org.junit.runner.RunWith;

@RunWith(EvoRunner.class) @EvoRunnerParameters(mockJVMNonDeterminism = true, useVFS = true, useVNET = true, resetStaticState = true, separateClassLoader = true, useJEE = true) 
public class JavaShell_ESTest extends JavaShell_ESTest_scaffolding {

  @Test(timeout = 4000)
  public void test0()  throws Throwable  {
    Future<?> future = executor.submit(new Runnable(){ 
            @Override public void run() { 
        try {
          JavaShell javaShell0 = new JavaShell();
          String[] stringArray0 = new String[16];
          try { 
            javaShell0.runMain("org.apache.groovy.uil.JavaShell$BytesJvaFileManager", "org.apache.groovy.uil.JavaShell$BytesJvaFileManager", stringArray0);
            fail("Expecting exception: SecurityException");
          
          } catch(SecurityException e) {
             //
             // Security manager blocks (\"java.lang.RuntimePermission\" \"closeClassLoader\")
             // java.lang.Thread.getStackTrace(Thread.java:1559)
             // org.evosuite.runtime.sandbox.MSecurityManager.checkPermission(MSecurityManager.java:434)
             // java.net.URLClassLoader.close(URLClassLoader.java:290)
             // com.sun.tools.javac.main.JavaCompiler.close(JavaCompiler.java:1714)
             // com.sun.tools.javac.main.JavaCompiler.close(JavaCompiler.java:1678)
             // com.sun.tools.javac.main.Main.compile(Main.java:573)
             // com.sun.tools.javac.api.JavacTaskImpl.doCall(JavacTaskImpl.java:129)
             // com.sun.tools.javac.api.JavacTaskImpl.call(JavacTaskImpl.java:138)
             // org.apache.groovy.util.JavaShell.compileAll(JavaShell.java:136)
             // org.apache.groovy.util.JavaShell.compile(JavaShell.java:108)
             // org.apache.groovy.util.JavaShell.runMain(JavaShell.java:94)
             // sun.reflect.GeneratedMethodAccessor45.invoke(Unknown Source)
             // sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
             // java.lang.reflect.Method.invoke(Method.java:498)
             // org.evosuite.testcase.statements.MethodStatement$1.execute(MethodStatement.java:257)
             // org.evosuite.testcase.statements.AbstractStatement.exceptionHandler(AbstractStatement.java:169)
             // org.evosuite.testcase.statements.MethodStatement.execute(MethodStatement.java:220)
             // org.evosuite.testcase.execution.TestRunnable.executeStatements(TestRunnable.java:307)
             // org.evosuite.testcase.execution.TestRunnable.call(TestRunnable.java:213)
             // org.evosuite.testcase.execution.TestRunnable.call(TestRunnable.java:55)
             // java.util.concurrent.FutureTask.run(FutureTask.java:266)
             // java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
             // java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
             // java.lang.Thread.run(Thread.java:748)
             //
             verifyException("org.evosuite.runtime.sandbox.MSecurityManager", e);
          }
        } catch(Throwable t) {
            // Need to catch declared exceptions
        }
      } 
    });
    future.get(4000, TimeUnit.MILLISECONDS);
  }

  @Test(timeout = 4000)
  public void test1()  throws Throwable  {
      JavaShell javaShell0 = new JavaShell();
      try { 
        javaShell0.runMain("q8<~Z2&4u{c*(", "", (String[]) null);
        fail("Expecting exception: GroovyRuntimeException");
      
      } catch(GroovyRuntimeException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("org.codehaus.groovy.tools.javac.MemJavaFileObject", e);
      }
  }

  @Test(timeout = 4000)
  public void test2()  throws Throwable  {
      JavaShell javaShell0 = new JavaShell();
      // Undeclared exception!
      try { 
        javaShell0.compileAll((String) null, "");
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
      }
  }

  @Test(timeout = 4000)
  public void test3()  throws Throwable  {
      JavaShell javaShell0 = new JavaShell();
      // Undeclared exception!
      try { 
        javaShell0.compile((String) null, (String) null);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
      }
  }

  @Test(timeout = 4000)
  public void test4()  throws Throwable  {
      JavaShell javaShell0 = new JavaShell((ClassLoader) null);
      // Undeclared exception!
      try { 
        javaShell0.compileAll("K1%mQ", "org.apache.groovy.io.StringBuilderWriter");
        fail("Expecting exception: GroovyRuntimeException");
      
      } catch(GroovyRuntimeException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("org.codehaus.groovy.tools.javac.MemJavaFileObject", e);
      }
  }

  @Test(timeout = 4000)
  public void test5()  throws Throwable  {
      JavaShell javaShell0 = new JavaShell();
      // Undeclared exception!
      try { 
        javaShell0.compile("]I|[|D6(h-,", "org.apache.groovy.util.JavaShell");
        fail("Expecting exception: GroovyRuntimeException");
      
      } catch(GroovyRuntimeException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("org.codehaus.groovy.tools.javac.MemJavaFileObject", e);
      }
  }

  @Test(timeout = 4000)
  public void test6()  throws Throwable  {
      ClassLoader classLoader0 = mock(ClassLoader.class, new ViolatedAssumptionAnswer());
      JavaShell javaShell0 = new JavaShell(classLoader0);
      String[] stringArray0 = new String[3];
      try { 
        javaShell0.runMain((String) null, (String) null, stringArray0);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
      }
  }
}
